utils/s3/credentials_providers/instance_profile_credentials_provider.cc

```diff
@@ -32,6 +32,8 @@ future<> instance_profile_credentials_provider::reload() {
     }
 }
 
+static constexpr auto EC2_SECURITY_CREDENTIALS_RESOURCE = "/latest/meta-data/iam/security-credentials";
```

Comment:
why not just 
`static constexpr auto EC2_SECURITY_CREDENTIALS_RESOURCE = "/latest/meta-data/iam/security-credentials/";`
and then
`req = http::request::make("GET", ec2_metadata_ip, EC2_SECURITY_CREDENTIALS_RESOURCE);`
and 
`req = http::request::make("GET", ec2_metadata_ip, seastar::format("{}{}", EC2_SECURITY_CREDENTIALS_RESOURCE, role));`
which results in just a single call to `format`?

Reply:
yes, it's simpler. but, IMHO, `seastar::format("{}{}", EC2_SECURITY_CREDENTIALS_RESOURCE, role)` is less readable than `seastar::format("{}/{}", EC2_SECURITY_CREDENTIALS_RESOURCE, role)` -- the latter implies that the resource and the role components are separated with "/".

Reply:
ok then